src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -69,4 +70,22 @@
     super(CONFIG, props, false);
   }
 
+  /**
+   * This method won't call {@link Configurable#configure(Map)} for {@link LiKafkaProducerConfig#AUDITOR_CLASS_CONFIG}
+   * during instance creation
+   */
+  @Override
+  public <T> T getConfiguredInstance(String key, Class<T> t) {
+    if (AUDITOR_CLASS_CONFIG.equals(key)) {
```

Comment:
Would this be too specific? I think always call `Configurable.configure()` is the right behavior for `getConfiguredInstance()`, The current problem is that we are not passing in the producer when we call `Configurable.configure()`. So I am thinking that maybe we just need to override `getConfiguredInstance` from `AbstractConfig` in LiKafkaProducerConfig` and make sure when `Configurable.configure()` is invoked,the configuration map with the underlying producer is passed in.